tests/integration/components/polaris-icon-test.js

```diff
@@ -44,19 +48,30 @@ test('it applies colors correctly', function(assert) {
     'purple'
   ];
 
+  assert.expect(2 + colors.length * 3);
+
   this.render(hbs`{{polaris-icon source="add" color=color}}`);
 
   // Check without any color set first.
   const icon = find(iconSelector);
-  assert.equal(icon.classList.length, 2, 'icon without color does not add color class');
+  assert.ok(icon.className.indexOf('Polaris-Icon--color') === -1, 'icon without color does not add color class');
+  assert.notOk(icon.classList.contains('Polaris-Icon--isColored'), 'icon without color does not add isColored class');
 
-  // Check all the available colors.
+  // Check all the available colors are handled correctly.
   for (const color of colors) {
     this.set('color', color);
 
     const colorClass = `Polaris-Icon--color${classify(color)}`;
     assert.ok(icon.classList.contains(colorClass), `icon with ${color} color applies ${colorClass} class`);
-    assert.equal(icon.classList.length, 3, `icon with ${color} color does not add other color classes`);
+
+    const colorClassNames = [...icon.classList].filter((className) => className.indexOf('Polaris-Icon--color') === 0);
```

Comment:
This doesn't bug me being all on one line, but for the record if I did this, you'd have me split it up 😛

Reply:
:trollface: 

Guess I thought that the intent of this line was more important than showing the details explicitly, but you're right, I should play by my own rules 😛